### PR TITLE
Log Event Label Attributes

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview

* Adds new feature to include labels as attributes on all log events.
* Adds new feature to attach global custom attributes to all log events.
* Adds new settings to configure this behavior.

### CODE REVIEW

**Review by hmstepanek — CHANGES_REQUESTED** [review]
Overall looks really good! Noted a couple things in comments.

File: newrelic/core/data_collector.py [hunk 1]
```diff
@@ -154,10 +156,59 @@ def send_dimensional_metric_data(self, start_time, end_time, metric_data):
...
+
+                common.update(custom_attributes)
+
+            # Add application labels as tags. prefixed attributes to common block
+            labels = self.configuration.labels
+            if not labels or not self.configuration.application_logging.forwarding.labels.enabled:
+                return common
+            elif not self.configuration.application_logging.forwarding.labels.exclude:
+                common.update({
+                    f"tags.{label['label_type']}": label['label_value']
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/core/data_collector.py:186]: Yes that limit is already are enforced in the function `_process_labels_setting`

File: tests/testing_support/validators/validate_log_event_collector_json.py [hunk 2]
```diff
@@ -33,36 +32,30 @@ def _validate_log_event_collector_json(wrapped, instance, args, kwargs):
...
-                for d in event:
-                    assert isinstance(d, dict)
+                # event is an array containing timestamp, level, message, attributes
+                assert len(event) == 4
+                assert isinstance(event["timestamp"], int)
+                assert isinstance(event["level"], str)
+                assert isinstance(event["message"], str)
+                assert isinstance(event["attributes"], dict)
+
+                expected_attribute_keys = sorted(
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply tests/testing_support/validators/validate_log_event_collector_json.py:55]: We talked about doing this originally, but there was some issue with that with multiple applications.

File: tests/agent_features/test_log_events.py [hunk 3]
```diff
@@ -396,3 +405,73 @@ def test():
...
+    session = list(txn.application._agent._applications.values())[0]._active_session
+
+    common = session.get_log_events_common_block()
+    # No labels should appear
+    assert common == {}
+
+
+# ================================================
+# Test Log Event Global Custom Attributes Settings
+# ================================================
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply tests/agent_features/test_log_events.py:464]: Done

**Review by hmstepanek — APPROVED** [review]

**Pull request merged**